File: files/2024-10-06/CSEC 06.10.R
library(tidyverse)
library(readxl)

path = "files/Excel Challenge October 6th.xlsx"
input1 = read_excel(path, range = "B2:C5") %>% mutate(id = 1)
input2 = read_excel(path, range = "E2:E6") %>% mutate(id = 1)
test  = read_excel(path, range = "G2:I14") 

result = input1 %>% 
  full_join(input2, by = "id") %>%
  select(Staff, PPE, Size)

all.equal(result, test)
# [1] TRUE

File: files/2024-10-06/CSEC 06.10.py
import pandas as pd

path = "files/Excel Challenge October 6th.xlsx"

input1 = pd.read_excel(path, usecols="B:C", skiprows=1, nrows=3).assign(id=1)
input2 = pd.read_excel(path, usecols="E", skiprows=1, nrows=4).assign(id=1)
test = pd.read_excel(path, usecols="G:I", skiprows=1, nrows=13).rename(columns=lambda x: x.replace('.1', ''))
result = pd.merge(input1, input2, on="id", how="outer")[['Staff', 'PPE', 'Size']]
print(result.equals(test)) # True